lp-lsp/lsp_base.ml

```diff
@@ -19,6 +19,8 @@ let parse_uri str =
 
 let mk_reply   r = `Assoc [ "jsonrpc", `String "2.0"; "id", `Int 0;        "result", `Assoc r ]
 let mk_event m p = `Assoc [ "jsonrpc", `String "2.0"; "method", `String m; "params", `Assoc p ]
+let mk_reply_null m = `Assoc [ "jsonrpc", `String "2.0"; "id", `Int 1; "result", `String m;]
```

Comment:
Oh, so that's a deeper problem. I had forgotten I had left the reply id as "future work".

Indeed we need to use the original request id in mk_reply, so that means we need to modify the code a bit and add an `~id` parameter to mk_reply.

Do you want to do it? Also do_initialize should be handled.

Reply:
Yes sure I can try to do it this afternoon